### PR TITLE
Feature->Develop | Added: Dynamic Environment Configurations

### PR DESCRIPTION
**Pull request by we-khushal-y001**
**Problem Statement**
1. Add dynamic environment configs from json file

**Describe The Fix/ Solution You Implemented**
1. I have added the json file in root asset folder and added the functionality to fetch the configs dynamically from json file using dart environment variable.

**Proof of your testing (Demo link or video links or image links)**
- For all the platforms
https://drive.google.com/file/d/1oidMT82515Gidzv0ZAt88PFtuxr78dYG/view?usp=sharing

**Dependencies**
Updated:
- 
Added:
- path of asset folder in pubspec.yaml file

### Merge Request Checklist

- [x] I have performed a self-review of my code
- [x] My code does follow `industry standards`
- [x] My code doesn't produce `warnings/ errors` for `dart analyzer`
- [ ] New and existing `tests` pass locally with my changes (No tests are there as of now)
- [ ] The code modified as part of this PR has been covered with tests
- [x] My code is `properly formatted`
- [x] I've followed the proper naming conventions
- [ ] I have rebased the `feature` on the latest `develop`
- [ ] I have run `flutter run` after the rebase
- [ ] I have added new dependencies/ updated old dependencies
- [x] I have read all the `comments` & `notes` in `wireframe` and verified that I did take care of that
- [ ] verify that the `UI` must match with `design` & `wireframe` if available
- [ ] I have updated the `version/ build` (`x.x.x+xxx`)
- [ ] I've verified that the latest commit of `develop` exists in your `feature` branch after rebase.
- [x] My fix achieves 100% of what is required
- [x] Can enhance the solution in the future because better implementation could be made

### CODE REVIEW

**Comment by we-prajapati-c001** [comment]
@we-khushal-y001 update PR title, use std format

File: lib/vaahextendflutter/services/logging_library/logging_library.dart [hunk 1]
```diff
@@ -80,24 +82,24 @@ class Log {
       Console.danger('$throwable\n$hint', data);
     }
     if (_config.enableCloudLogs && !disableCloudLogging) {
-      final hintWithData = {
-        'hint': hint,
-        'data': data,
+      final hintWithData = <String, Object>{
+        'hint': hint ?? {},
+        'data': data ?? {},
```
Comment (we-khushal-y001) [inline lib/vaahextendflutter/services/logging_library/logging_library.dart]: @we-khushal-y001 to remove logging service changes from this PR.

Comment (we-chintan-p001) [reply lib/vaahextendflutter/services/logging_library/logging_library.dart]: Please resolve

File: lib/vaahextendflutter/helpers/assets.dart [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class DataAssets {
+  const DataAssets._();
+
+  static String envJson(String environment) => 'assets/env/$environment.json';
+}
```
Comment (we-chintan-p001) [inline lib/vaahextendflutter/helpers/assets.dart]: no need for this file now i guess

File: lib/vaahextendflutter/services/logging_library/logging_library.dart (hunk 1 above)
Comment (we-chintan-p001) [inline lib/vaahextendflutter/services/logging_library/logging_library.dart]: revert as it was, will be handled in https://github.com/webreinvent/vaahflutter/pull/38

File: lib/vaahextendflutter/services/logging_library/logging_library.dart [hunk 3]
```diff
@@ -80,24 +82,24 @@ class Log {
...
       };
       for (final service in _services) {
         switch (service) {
           case SentryLoggingService:
             SentryLoggingService.logException(
               throwable,
               stackTrace: stackTrace,
-              hint: hintWithData,
+              hint: Hint.withMap(hintWithData),
```
Comment (we-chintan-p001) [inline lib/vaahextendflutter/services/logging_library/logging_library.dart]: revert, will be handled in https://github.com/webreinvent/vaahflutter/pull/38

File: lib/vaahextendflutter/services/logging_library/logging_library.dart [hunk 4]
```diff
@@ -80,24 +82,24 @@ class Log {
...
             );
             return;
           case FirebaseLoggingService:
             FirebaseLoggingService.logException(
               throwable,
               stackTrace: stackTrace,
-              hint: hintWithData,
+              hint: Hint.withMap(hintWithData),
```
Comment (we-chintan-p001) [inline lib/vaahextendflutter/services/logging_library/logging_library.dart]: revert, will be handled in https://github.com/webreinvent/vaahflutter/pull/38

**Review by we-chintan-p001 — CHANGES_REQUESTED** [review]

File: lib/vaahextendflutter/helpers/assets.dart (hunk 2 above)
Comment (we-khushal-y001) [reply lib/vaahextendflutter/helpers/assets.dart]: done

File: lib/vaahextendflutter/services/logging_library/logging_library.dart (hunk 1 above)
Comment (we-khushal-y001) [reply lib/vaahextendflutter/services/logging_library/logging_library.dart]: done

File: lib/vaahextendflutter/services/logging_library/logging_library.dart (hunk 3 above)
Comment (we-khushal-y001) [reply lib/vaahextendflutter/services/logging_library/logging_library.dart]: done

File: lib/vaahextendflutter/services/logging_library/logging_library.dart (hunk 4 above)
Comment (we-khushal-y001) [reply lib/vaahextendflutter/services/logging_library/logging_library.dart]: done

**Review by we-chintan-p001 — APPROVED** [review]

File: lib/vaahextendflutter/env/env.dart [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+  factory EnvironmentConfig.defaultConfig() {
+    return EnvironmentConfig(
+      appTitle: 'VaahFlutter',
+      appTitleShort: 'VaahFlutter',
+      envType: 'default',
+      version: '1.0.0',
+      build: '1',
+      apiUrl: '',
+      timeoutLimit: 20 * 1000, // 20 seconds
```
Comment (themodernpk) [inline lib/vaahextendflutter/env/env.dart]: @we-khushal-y001 @we-prajapati-c001  write `20000, // 20 * 1000ms = 20 seconds`, why to waste compute on simple multiplication.

File: lib/vaahextendflutter/env/env.dart [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+      enableLocalLogs: true,
+      enableCloudLogs: true,
```
Comment (themodernpk) [inline lib/vaahextendflutter/env/env.dart]: @we-khushal-y001 @we-prajapati-c001 this should be false by default

File: lib/vaahextendflutter/env/env.dart [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      enableApiLogInterceptor: true,
```
Comment (themodernpk) [inline lib/vaahextendflutter/env/env.dart]: @we-khushal-y001 @we-prajapati-c001 if api url is not set, do we need to keep it as true?

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: lib/vaahextendflutter/env/env.dart (hunk 5 above)
Comment (we-chintan-p001) [reply lib/vaahextendflutter/env/env.dart]: makes sense, agreed

File: lib/vaahextendflutter/env/env.dart (hunk 6 above)
Comment (we-chintan-p001) [reply lib/vaahextendflutter/env/env.dart]: makes sense, agreed on this too.

File: lib/vaahextendflutter/env/env.dart (hunk 7 above)
Comment (we-chintan-p001) [reply lib/vaahextendflutter/env/env.dart]: hm, there are two urls which we fetch let's say 
1. https://vaah.dev/cms which is our base API URL
2. https://jsonplaceholder.typicode.com/todos/1 is another we fetch which is absolute URL we wanna fetch

so to log for all type of API calls

Comment (we-chintan-p001) [reply lib/vaahextendflutter/env/env.dart]: also we can not initialize a variable depending on some other variable from same constructor

File: lib/vaahextendflutter/env/env.dart (hunk 5 above)
Comment (we-khushal-y001) [reply lib/vaahextendflutter/env/env.dart]: done sir

File: lib/vaahextendflutter/env/env.dart (hunk 6 above)
Comment (we-khushal-y001) [reply lib/vaahextendflutter/env/env.dart]: done sir

File: .vscode/launch.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
+{
+    "version": "0.2.0",
+    "inputs": [
+        {
+            "id": "option",
+            "type": "pickString",
+            "description": "Select option",
```
Comment (we-chintan-p001) [inline .vscode/launch.json]: rename this variable to build mode, everywhere.

**Review by we-chintan-p001 — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 8 above)
Comment (we-khushal-y001) [reply .vscode/launch.json]: done

**Pull request merged**